drivers/sdhc/sdhc_max32.c

```diff
@@ -0,0 +1,416 @@
+	const struct pinctrl_dev_config *pcfg;
+	unsigned int power_delay_ms;
+	unsigned int bus_volt;
+	const struct device *clock;
+	struct max32_perclk perclk;
+};
+
+static void sdhc_max32_init_props(const struct device *dev)
+{
+	struct sdhc_max32_data *sdhc_data = dev->data;
+	const struct sdhc_max32_config *sdhc_config = dev->config;
+
+	memset(sdhc_data, 0, sizeof(struct sdhc_max32_data));
+
+	sdhc_data->props.f_min = SDHC_CLOCK / (SDHC_SDHC_PCLK_DIV * SDHC_SDHC_MAX_DIV_VAL);
+	sdhc_data->props.f_max = SDHC_CLOCK;
+	sdhc_data->props.is_spi = 0;
+	sdhc_data->props.max_current_180 = 0;
+	sdhc_data->props.max_current_300 = 0;
+	sdhc_data->props.max_current_330 = 0;
+	sdhc_data->props.host_caps.timeout_clk_freq = 0x01;
+	sdhc_data->props.host_caps.timeout_clk_unit = 1;
+	sdhc_data->props.host_caps.sd_base_clk = 0x00;
+	sdhc_data->props.host_caps.max_blk_len = 0b10;
+	sdhc_data->props.host_caps.bus_8_bit_support = false;
+	sdhc_data->props.host_caps.bus_4_bit_support = false;
+	sdhc_data->props.host_caps.adma_2_support = true;
+	sdhc_data->props.host_caps.high_spd_support = true;
+	sdhc_data->props.host_caps.sdma_support = true;
+	sdhc_data->props.host_caps.suspend_res_support = true;
+	sdhc_data->props.host_caps.vol_330_support = true;
+	sdhc_data->props.host_caps.vol_300_support = true;
+	sdhc_data->props.host_caps.vol_180_support = true;
+	sdhc_data->props.host_caps.address_64_bit_support_v4 = false;
+	sdhc_data->props.host_caps.address_64_bit_support_v3 = false;
+	sdhc_data->props.host_caps.sdio_async_interrupt_support = true;
+	sdhc_data->props.host_caps.slot_type = 00;
+	sdhc_data->props.host_caps.sdr50_support = true;
+	sdhc_data->props.host_caps.sdr104_support = true;
+	sdhc_data->props.host_caps.ddr50_support = true;
+	sdhc_data->props.host_caps.uhs_2_support = false;
+	sdhc_data->props.host_caps.drv_type_a_support = true;
+	sdhc_data->props.host_caps.drv_type_c_support = true;
+	sdhc_data->props.host_caps.drv_type_d_support = true;
+	sdhc_data->props.host_caps.retune_timer_count = 0;
+	sdhc_data->props.host_caps.sdr50_needs_tuning = 0;
+	sdhc_data->props.host_caps.retuning_mode = 0;
+	sdhc_data->props.host_caps.clk_multiplier = 0;
+	sdhc_data->props.host_caps.adma3_support = false;
+	sdhc_data->props.host_caps.vdd2_180_support = false;
+	sdhc_data->props.host_caps.hs200_support = false;
+	sdhc_data->props.host_caps.hs400_support = false;
+	sdhc_data->props.power_delay = sdhc_config->power_delay_ms;
+}
+
+static int sdhc_max32_init(const struct device *dev)
+{
+	const struct sdhc_max32_config *sdhc_config = dev->config;
+	int ret = 0;
+	mxc_sdhc_cfg_t cfg;
+
+	ret = pinctrl_apply_state(sdhc_config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("Pinctrl apply error:%d", ret);
+		return ret;
+	}
+
+	ret = clock_control_on(sdhc_config->clock, (clock_control_subsys_t)&sdhc_config->perclk);
+	if (ret) {
+		LOG_ERR("Clock control on error:%d", ret);
+		return ret;
+	}
+
+	cfg.bus_voltage = sdhc_config->bus_volt;
```

Comment:
This *cannot* be configured statically. If memory serves, the SDHC spec specifies cards should start at 3.3V, and then switch to 1.8V during initialization. So there are two options here:
- support dynamic voltage configuration
- only support cards at 3.3V

Have you been able to initialize a card at 1.8V using this mechanism? Maybe my memory is wrong, but I thought that only eMMC could boot at 1.8V. SD cards need to switch to 1.8V at runtime